### PR TITLE
refactor: allow start from project root

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

allow `[npm/pnpm/yarn] start` from project root by:

- add `--from-root` option to change working directory
- adjust UI proxy

also clean up unnecessary code when reading private key

## Linear Issue Reference

LOG-2259

## Testing

start from project root, AC ok

image

### CODE REVIEW

File: packages/core/src/env-set/oidc.ts [hunk 1]
```diff
@@ -6,11 +6,11 @@ import inquirer from 'inquirer';
 
 import { noInquiry } from './parameters';
 
-const readPrivateKey = async (path: string): Promise<string> => {
+const readPrivateKey = async (): Promise<string> => {
```
Comment (simeng-li) [inline packages/core/src/env-set/oidc.ts:9]: finally got what did you mean by removing useless path input  🤣

**Review by simeng-li — APPROVED** [review]
LGTM

**Review by wangsijie — APPROVED** [review]

**Comment by demonzoo** [comment]
Just curious: What's the difference between `pnpm start --from-root` and `pnpm-dev` then? I think both serve the purpose of starting the dev server?

File: packages/core/src/env-set/oidc.ts (hunk 1 above)
Comment (gao-sun) [reply packages/core/src/env-set/oidc.ts:9]: hahah

**Review by gao-sun — COMMENTED** [review]
discussed offline with Charles

**Review by demonzoo — APPROVED** [review]

**Pull request merged**